### PR TITLE
Add elementwise and broadcast simulator tests

### PR DESCRIPTION
**Pull request by Hardcode84**
As I'm using `torch.tensor` in simulator, all elementwise ops and broadcasting should work out of the box.

### CODE REVIEW

File: tests/kernel/wave/wave_sim_test.py [hunk 1]
```diff
@@ -6,6 +6,152 @@
...
+    # Other hyperparameters
+    LOAD_ELEMS_PER_THREAD = tkl.sym.LOAD_ELEMS_PER_THREAD
+    STORE_ELEMS_PER_THREAD = tkl.sym.STORE_ELEMS_PER_THREAD
+
+    # Expose user-constraints
+    constraints: list[tkw.Constraint] = [tkw.WorkgroupConstraint(M, BLOCK_M, 0)]
+    constraints += [tkw.WorkgroupConstraint(N, BLOCK_N, 1)]
+
+    constraints += [
+        tkw.HardwareConstraint(threads_per_wave=64, waves_per_block=(1, 1, 1))
```
Comment (harsh-amd) [inline tests/kernel/wave/wave_sim_test.py:27]: I added a vector_shapes arg which specifies the "intrinsic" tile size that will probably be useful for the simulator when we simulate things at that level. Right now, I think the simulator ignores that, but just fyi.

File: tests/kernel/wave/wave_sim_test.py [hunk 2]
```diff
@@ -6,6 +6,152 @@
...
+
+    a = torch.randn(128, 256, dtype=torch.float32)
+    b = torch.randn(128, 256, dtype=torch.float32)
+    c = torch.zeros(128, 256, dtype=torch.float32)
+    eltwise(a, b, c)
+    assert_allclose(c, a + b)
+
+
+def test_broadcast_1():
+    # Input sizes
```
Comment (harsh-amd) [inline tests/kernel/wave/wave_sim_test.py:48]: Nice!

**Review by harsh-amd — APPROVED** [review]
Thanks for adding these tests. Great that broadcasting just works out of the box.

File: tests/kernel/wave/wave_sim_test.py (hunk 1 above)
Comment (Hardcode84) [reply tests/kernel/wave/wave_sim_test.py:27]: Yeah, at this point simulator completely ignores the `constraints` arg

**Review by harsh-nod — APPROVED** [review]
Thanks! LGTM!

**Pull request merged**